### DISCUSSION TITLE
Why M2A field returns null when being fetched via GraphQL API?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#10292] [Archive]
I have established a M2A relationship as shown below:

1. Collection view:

![image](https://user-images.githubusercontent.com/42278237/144696561-7848b8e0-fcdd-49f0-9891-2d7892e721d6.png)

2. Single record view:
![image](https://user-images.githubusercontent.com/42278237/144696566-b3a17339-36e8-4b06-88d8-578f6e43c3d6.png)

However, when fetching the M2A field via Directus' GraphQL API, the returned value is `null` as shown below:

![image](https://user-images.githubusercontent.com/42278237/144696667-d7fda731-98af-4993-b755-abd7ee9aec01.png)

Moreover, the type of the M2A field is `String` as shown below:

![image](https://user-images.githubusercontent.com/42278237/144696686-1d87e744-9fe5-4941-b92d-3cb3f0da0d67.png)

But how do I fetch the column in the next level if it is type is a scalar type?

For example, if the `M2A` field references to a collection `video_asset`, I think it is **impossible** to fetch column `video_asset.url` (not the PK) if `media` is of type `String`, is it?

Hope someone can help me.

Thanks in advance!

**Comment by joselcvarela** [comment]
Hello @powerseed 
What version of Directus are you using? With latest version my M2A field has the proper return type which is the junction type.
Also I can properly query things from this field.
As a friendly reminder, don't forget you would need to use fragments with M2A relations https://docs.directus.io/reference/introduction/#graphql

**Reply by powerseed** [reply]
Hi @joselcvarela 

I am using the latest version, 9.1.2

I cannot use fragments because the M2A field is of type `String`.

It is a scalar so when I use fragments it says that "String doesn't have a subfield".

Do you see any issues in the screenshots I posted?

Thanks in advance!

**Reply by rijkvanzanten** [reply]
@powerseed Does the user you're using the GraphQL endpoint with have the correct read permissions to the related collection(s)? (https://github.com/directus/directus/issues/8463)

**Reply by powerseed** [reply]
@rijkvanzanten 

Yes I think so. It is an admin role.

Besides, if the user doesn't have the permission, I would not be able to see the referenced table (in my case it is called `video_asset`) in GUI, is that correct?

But in my 2nd screenshot I can see it and select records from it.

**Reply by rijkvanzanten** [reply]
It's hard to tell exactly without having your schema in front of me.. What is the exact M2A relationship setup? Could it be `media` isn't the relational field in this example? From your other Discussion, I recall that the name of the related item was `video_asset` (assuming you're working with the same schema here 🤔 )

**Reply by powerseed** [reply]
@rijkvanzanten 

The M2A relationship is as follows:

`organization_content` has a M2A field called `media`, and this field references two tables, `directus_files` and `video_asset`.

It works fine in GUI. I can select records from both of the tables, `directus_files` and `video_asset`, for the `media` M2A field.

But when fetching via GraphQL API, the `media` M2A field always returns `null`.

I will keep digging. 

If there is anything you want me to check, please just let me know.

Thank you so much!

**Reply by powerseed** [reply]
@rijkvanzanten and @joselcvarela 

It is my bad again...

I thought I was using an admin account but I was not...

After I assigned the correct permissions, it works perfectly.

I cannot apologize more.

And thank you for all the help!
